src/policy.rs

```diff
@@ -114,6 +120,39 @@ impl AccessPolicy {
         })
     }
 
+    /// Convert policy to integer value (for comparison).
+    /// Each attribute is mapped to an integer value and the algebraic
+    /// expression is applied with those values.
+    /// We must keep a mapping of each attribute to the corresponding integer
+    /// value in order to avoid having 2 different attributes with same integer
+    /// value
+    fn to_u32(&self, attribute_mapping: &mut HashMap<Attribute, u32>) -> u32 {
+        match self {
+            AccessPolicy::Attr(attr) => {
+                if let Some(integer_value) = attribute_mapping.get(attr) {
+                    debug!("found attribute: {:?}, value = {}", attr, *integer_value);
+                    *integer_value
+                } else {
+                    // To assign an integer value to a new attribute, we take the current max
+                    // integer value + 1
+                    let max_value = attribute_mapping.values().max();
+                    let max = if let Some(max) = max_value {
```

Comment:
```suggestion
                    let max = max_value.map(|max| *max + 1).unwrap_or(1);
```

just a suggestion

Reply:
it could even be merged with the previous line (as we don't use `max_value` later on)

Reply:
thank you, I'll take the 2 suggestions